src/test/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapperTest.java

```diff
@@ -132,37 +149,70 @@ void givenNoOrganizationAccessWhenApplyThenOk(){
                 .roles(Set.of("ROLE"))
                 .email("EMAIL")
                 .build());
-
-        testApplyOk(iamUserInfo, user, organizationRoles);
+        
+        UserInfo expected = UserInfo.builder()
+          .fiscalCode("FISCALCODE")
+          .familyName("FAMILYNAME")
+          .name("NAME")
+          .issuer("ISSUER")
+          .build();
+
+        testApplyOk(iamUserInfo, user, organizationRoles, expected);
     }
 
-    private void testApplyOk(IamUserInfoDTO iamUserInfo, User user, List<Operator> organizationRoles) {
-        Mockito.when(usersRepositoryMock.findById(iamUserInfo.getInnerUserId())).thenReturn(Optional.of(user));
-        Mockito.when(operatorsRepositoryMock.findAllByUserId(user.getUserId())).thenReturn(organizationRoles);
+    @Test
+    void givenSystemUserWhenApplyThenOk(){
+        // Given
+        IamUserInfoDTO iamUserInfo = IamUserInfoDTO.builder()
+          .systemUser(Boolean.TRUE)
+          .userId("EXTERNALUSERID")
+          .innerUserId("INNERUSERID")
+          .fiscalCode("FISCALCODE")
+          .familyName("FAMILYNAME")
+          .name("NAME")
+          .issuer("IPA_CODE")
+          .organizationAccess(IamUserOrganizationRolesDTO.builder()
+            .organizationIpaCode("IPA_CODE")
+            .roles(Collections.singletonList(Constants.ROLE_ADMIN))
+            .build())
+          .build();
+
+        UserInfo expected = UserInfo.builder()
+          .userId("EXTERNALUSERID")
+          .mappedExternalUserId("IPA_CODE-WS_USER")
+          .fiscalCode("FISCALCODE")
+          .familyName("FAMILYNAME")
+          .name("NAME")
+          .issuer("IPA_CODE")
+          .organizations(Collections.singletonList(UserOrganizationRoles.builder()
+            .organizationIpaCode("IPA_CODE")
+            .roles(Collections.singletonList(Constants.ROLE_ADMIN))
+            .build()))
+          .build();
+
+        testApplyOk(iamUserInfo, null, null, expected);
+    }
 
+    private void testApplyOk(IamUserInfoDTO iamUserInfo, User user, List<Operator> organizationRoles, UserInfo expected) {
+
+        if (!iamUserInfo.isSystemUser()) {
+            Mockito.when(usersRepositoryMock.findById(iamUserInfo.getInnerUserId())).thenReturn(Optional.of(user));
+            Mockito.when(operatorsRepositoryMock.findAllByUserId(user.getUserId())).thenReturn(organizationRoles);
+            expected.setUserId(user.getUserId());
+            expected.setMappedExternalUserId(user.getMappedExternalUserId());
+            expected.setOrganizations(organizationRoles.stream()
+              .map(r -> UserOrganizationRoles.builder()
+                .operatorId(r.getOperatorId())
+                .organizationIpaCode(r.getOrganizationIpaCode())
+                .roles(new ArrayList<>(r.getRoles()))
+                .email(r.getEmail())
+                .build())
+              .toList());
```

Comment:
even if it's not related to your task, please set also these fields with fixed values in the respective invocations.

This method is becoming useless... I think it could be better to remove it